sys/net/routing/aodvv2/aodv.c

```diff
@@ -348,54 +346,100 @@ static void *_aodv_receiver_thread(void *arg)
-                           .sin6_port = HTONS(MANET_PORT)
-                         };
+    /* start server (which means registering AODVv2 receiver for the chosen port) */
+    server.pid = sched_active_pid; /* sched_active_pid is our pid, since we are currently act */
+    server.demux_ctx = (uint32_t)HTONS(MANET_PORT);
+    ng_netreg_register(NG_NETTYPE_UDP, &server);
 
-    int sock_rcv = socket_base_socket(PF_INET6, SOCK_DGRAM, IPPROTO_UDP);
+    while (1) {
+        msg_receive(&msg);
 
-    if (-1 == socket_base_bind(sock_rcv, &sa_rcv, sizeof(sa_rcv))) {
-        DEBUG("Error: bind to receive socket failed!\n");
-        socket_base_close(sock_rcv);
-        return NULL;
+        switch (msg.type) {
+            case NG_NETAPI_MSG_TYPE_RCV:
+                AODV_DEBUG("received data:");
+                ng_pktsnip_t *pkt = ((ng_pktsnip_t *)msg.content.ptr);
+                if (pkt->size <= UDP_BUFFER_SIZE) {
+                    memcpy(buf_rcv, pkt->data, pkt->size);
+
+                    if(pkt->next->next->type == NG_NETTYPE_IPV6) {
+                        ipv6_addr_t_to_netaddr(&(((ng_ipv6_hdr_t*)(pkt->next->next->data))->src), &_sender);
+                    }
+
+                    if (netaddr_cmp(&_sender, &na_local) == 0) {
+                        AODV_DEBUG("received our own packet, dropping it.\n");
+                    }
+                    else {
+                        aodv_packet_reader_handle_packet((void *) buf_rcv, pkt->size, &_sender);
+                    }
+                }
+                ng_pktbuf_release(pkt);
+                break;
+            case NG_NETAPI_MSG_TYPE_SND:
+                break;
+            case NG_NETAPI_MSG_TYPE_GET:
+            case NG_NETAPI_MSG_TYPE_SET:
+                msg_reply(&msg, &reply);
+                break;
+            default:
+                AODV_DEBUG("received something unexpected");
+                break;
+        }
     }
 
-    AODV_DEBUG("ready to receive data\n");
-    while (true) {
-        int32_t rcv_size = socket_base_recvfrom(sock_rcv, (void *)buf_rcv, UDP_BUFFER_SIZE, 0,
-                                        &sa_rcv, &fromlen);
-
-        if (rcv_size < 0) {
-            AODV_DEBUG("ERROR receiving data!\n");
-        }
+    return NULL;
+}
 
-        AODV_DEBUG("_aodv_receiver_thread() %s:",
-              ipv6_addr_to_str(addr_str, IPV6_MAX_ADDR_STR_LEN, &_v6_addr_local));
-        DEBUG(" UDP packet received from %s\n",
-              ipv6_addr_to_str(addr_str, IPV6_MAX_ADDR_STR_LEN, &sa_rcv.sin6_addr));
+/**
+    The code o this function was borrowed from:
+    examples/ng_networking/udp.c:34 - :90
+*/
+static void send(ng_ipv6_addr_t addr, uint16_t port, void *data, size_t data_length)
+{
+    ng_pktsnip_t *payload, *pkt_with_udp, *pkt_with_ip;
 
-        struct netaddr _sender;
-        ipv6_addr_t_to_netaddr(&sa_rcv.sin6_addr, &_sender);
+    /* convert to correct byteorder */
+    port = HTONS(port);
```

Comment:
so removing all instances of HTONS in https://github.com/Lotterleben/RIOT/commit/4d95d13888c3fb040a1fe8dca6630e4205f0d7ad was a bit overeager.. ^^